core/src/wallet/tezos/TezosLikeWallet.cpp

```diff
@@ -68,6 +68,10 @@ namespace ledger {
             _observer = observer;
             _keychainFactory = keychainFactory;
             _synchronizerFactory = synchronizer;
+            // TODO This is a dirty patch because Tezos uses the AbstractBlockchainExplorer and synchronizer
+            // Most of the code of these classes were made for BTC and offer very few advantage for simpler
+            // currencies (mainly it avoid rewriting a reorg case)
+            configuration->putBoolean(api::Configuration::DEACTIVATE_SYNC_TOKEN, false);
```

Comment:
Hm, seems fine to me but do we have a better plan to clean this?

Reply:
Either we can fine tune AbstractBlockchainExplorer to handle this through inheritance or configuration. Which has the advantage of avoiding rewriting lot of stuffs.

Or we remove AbstractBlockchainExplorer and rewrite some parts to make this class standalone. I'm more for rewriting, because the benefits of having a common ground for BTC, ETH, XTZ is not very high and explorer/synchronizer becomes difficult to read even when XTZ synchronization seems to be pretty straightforward.